### PR TITLE
Flatten & Intersect group membership in Slick

### PR DESCRIPTION
**Pull request by MatthewBemis**
For realms in MySQL
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala [hunk 1]
```diff
@@ -137,7 +137,53 @@ trait RawlsGroupComponent {
...
+
+    private def listGroupUsers(groupRec: RawlsGroupRecord): ReadAction[Set[RawlsUserRef]] = {
+      findUsersByGroupName(groupRec.groupName).result flatMap { userRecs =>
+        DBIO.successful(userRecs.map(u => RawlsUserRef(RawlsUserSubjectId(u.userSubjectId))).toSet)
+      }
+    }
+
+    def flattenGroupMembership(groupRef: RawlsGroupRef): ReadAction[Set[RawlsUserRef]] = {
+      load(groupRef) flatMap { groupOption =>
+        groupOption match {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala]: can remove `groupOption =>  groupOption match {` here.  Flatmaps can go directly to cases.

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala [hunk 2]
```diff
@@ -137,7 +137,53 @@ trait RawlsGroupComponent {
         
         nextLevelUp.result.flatMap { nextGroups =>
           val nextCumulativeGroups = cumulativeGroups ++ nextGroups
-          listParentGroupsRecursive(nextGroups.toSet -- nextCumulativeGroups, nextCumulativeGroups)
+          listParentGroupsRecursive(nextGroups.toSet -- cumulativeGroups, nextCumulativeGroups)
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala:140]: was this change intentional?  was it tested?

**Comment by dvoet** [comment]
:+1:

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala (hunk 2 above)
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala:140]: yes, this was intentional. it was a bug that caused the recursion to terminate after the first step. it's out of scope for this PR but i can add a test for this function if you'd like

Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala:140]: The logic looks good.  No need to add tests for this PR.  I added a story: https://broadinstitute.atlassian.net/browse/GAWB-419

**Comment by jmthibault79** [comment]
:+1:
